waftools/detections/compiler.py

```diff
@@ -13,7 +13,8 @@ def __get_cc_env_vars__(cc):
 
 def __test_and_add_flags__(ctx, flags):
     for flag in flags:
-        ctx.check_cc(cflags=flag, uselib_store="compiler", mandatory=False)
+        if ctx.check_cc(cflags='-Werror ' + flag, mandatory=False):
+            ctx.env.CFLAGS += [flag]
     ctx.env.CFLAGS += ctx.env.CFLAGS_compiler
```

Comment:
since you are removing `uselib_store="compiler"`, I think you should be able to safely remove the `ctx.env.CFLAGS += ctx.env.CFLAGS_compiler` line

Reply:
I did consider that, but didn't know whether or not `ctx.env.CFLAGS_compiler` could already have some values in it from some prior checks, and together with my unfamiliarity in general with `waf`, I decided to keep it.

If you think it's safe to remove, I got no problem with removing it.

Reply:
Following discussion on  IRC with @pigoz, I removed the `ctx.env.CFLAGS_compiler` thingy and force pushed.